April tag

wilsomc opened the pull request.

wilsomc committed:
WIP mot working on hannah robot will test on ryker

wilsomc committed:
something

wilsomc committed:
Merge branch 'main' into April-tag

wilsomc committed:
something is happening but not quite

wilsomc committed:
working can find limelight

wilsomc committed:
added variables

wilsomc committed:
testing gyro

wilsomc committed:
Merge branch 'April-tag'

wilsomc committed:
Merge branch 'main' of https://github.com/gwhs/2024v2

wilsomc committed:
second drivetrain added

RobertTheBlair commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -50,4 +50,19 @@ public static class OperatorConstants
      public static final int kDriverControllerPort = 0;
   }
+
+  public static final class LimeLightConstants {
```
Might want to put the limelight constants in a different file.  They will vary by which robot is being used,

wilsomc replied:
will do in a different branch

RobertTheBlair commented on src/main/java/frc/robot/subsystems/LimeVision/LimeLightSub.java:
```diff
@@ -0,0 +1,111 @@
+    SmartDashboard.putNumber("theta", getTheta());
+    SmartDashboard.putNumber("AngleToTarget", getAngle());
+
+    if (getTx() < 0.5 && getTx() > -0.5)
```
This testing code needs to be removed before we merge.

RobertTheBlair commented on src/main/java/frc/robot/subsystems/LimeVision/PIDMove.java:
```diff
@@ -0,0 +1,38 @@
+
+// private LimeLightSub limeLightSub = new LimeLightSub("limelight");
+
+  public PIDMove(SwerveSubsystem swerveSystem, LimeLightSub limeLightSub, double kP, double Ki, double kD, double setPoint) {
```
This is more like a Command class - where it interacts with two subsystems, and uses them to achieve a goal.
At the moment, it is effectively doing nothing

RobertTheBlair commented on src/main/java/frc/robot/testcontainers/DriveContainer.java:
```diff
@@ -40,7 +40,7 @@ public class DriveContainer implements BaseContainer
 
   public String getDriveTrainName(){
-    return "swerve/ryker_falcon";
+    return "swerve/vision";
```
Discuss with Chris if we should merge this.   I believe we don't want do that yet.

RobertTheBlair commented on src/main/java/frc/robot/testcontainers/VisionContainer.java:
```diff
@@ -3,13 +3,17 @@
     private final CommandXboxController m_driverController =
         new CommandXboxController(OperatorConstants.kDriverControllerPort);
 
+    private LimeLightSub limeLightSub = new LimeLightSub("limelight");
```
Once you have debugged the drive setup, you can add the drive subsystem here, and override the getDriveTrainName function to return "swerve/vision"

RobertTheBlair reviewed (changes requested):
A few changes are needed - once we have the drivetrain working in vision robot, restore the name in DriveContainer, and add the swerve subsystem to VisionContainer

wilsomc committed:
Add files via upload

Code that I might try implementing see if it works, need to find a way to create swerve subsystem to vision container. at the moment using drive container because everything is already set up

wilsomc committed:
Add files via upload

Testing autoAlign, using driveContainer because easier as I don't have to add swerve subsystem into vision container. will do in future but i need time to figure it out

wilsomc committed:
it sometimes work

wilsomc committed:
cleaned up code

wilsomc committed:
can follow the April Tag

wilsomc committed:
Merge branch 'April-tag' of https://github.com/gwhs/2024v2 into April-tag

wilsomc committed:
Merge branch 'main' of https://github.com/gwhs/2024v2

wilsomc committed:
working

RobertTheBlair commented on src/main/java/frc/robot/subsystems/LimeVision/LimeLightSub.java:
```diff
@@ -0,0 +1,125 @@
+
+    if (getTx() < 0.5 && getTx() > -0.5)
+    {
+      System.out.println(getError());
```
put this on smart dashboard

wilsomc committed:
corrections reviewed

wilsomc committed:
corrections reviewed
